### DISCUSSION TITLE
flashing NSPanel with HA core (hence no HA addons available)

### DISCUSSION THREAD

**Discussion by k1l** [Blackymas/NSPanel_HA_Blueprint#2444] [Q&A] [answered]
Hi, i am using HA core so i cant use espHome addon. But since the whole Howto and all other Blogs and Videos only focus on using the EspHome Addon i am confused on how to flash it using the terminal esphome tool.

So, what files do i actually need to flash? (nspanel_esphome.yaml, nspanel_blueprint.yaml?) And do those .yaml files actually work without using the addon?
Everyone is referencing a nspanel(_eu).tft but thats not existent in the github repo anymore.

**Comment by edwardtfn** [comment, marked as answer]
The process should be very similar, as you can have the ESPHome Dashboard even when not running it as an add-on.
But if you don't have the dashboard and is using only command line, copy the content of that yaml on [Step 8 of installation instructions](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/install.md#setting-up-a-device-on-esphome-dashboard) into a local file and edit the device name and wifi credentials on substitutions just like the instructions there, then build the binaries using the command line.
You will use the `.bin` file in ESPTool or ESPHome Web to flash your device.

**Reply by edwardtfn** [reply]
Sorry, I'm typing on my phone so I don't have so many resources to share examples now. But anyways, everything else should be just like described there.

**Reply by k1l** [reply]
Alright, thank you very much for the reply.
I managed to get it flashed. I collected some stuff on the way that might be helpfull to others, too: 
You can run the esphome dashboard https://esphome.io/guides/getting_started_command_line#bonus-esphome-dashboard with `esphome dashboard /path/to/some/folder/ --open-ui` but flashing only works when using some chrome based browser.

the tft file is not needed anymore since that is inlcuded in the homeassistant device section now.

only issue so far is, that i used the cover addon and even tho i hear the relay clicking it doesnt work (it used to work with stock firmware before.).

**Reply by k1l** [reply]
Ok, i found the issue: the duration for opening and closing the cover on the devices page was still set to 1ms, which equals disabled. I was just confused because you could hear the relay click.
It works now. Thanks!
